Dapper/SqlMapper.DapperRow.cs

```diff
@@ -208,6 +209,41 @@ private object SetValue(string key, object value, bool isAdd)
             }
 
             #endregion
+
+
+            #region Implementation of IReadOnlyDictionary<string,object>
+
+
+            int IReadOnlyCollection<KeyValuePair<string, object>>.Count
+            {
+                get
+                {
+                    return values.Count(t => !(t is DeadValue));
+                }
+            }
+
+            bool IReadOnlyDictionary<string, object>.ContainsKey(string key)
+            {
+                int index = table.IndexOfName(key);
+                return index >= 0 && index < values.Length && !(values[index] is DeadValue);
+            }
+
+            object IReadOnlyDictionary<string, object>.this[string key]
+            {
+                get { TryGetValue(key, out object val); return val; }
+            }
+
+            IEnumerable<string> IReadOnlyDictionary<string, object>.Keys
+            {
+                get { return this.Select(kv => kv.Key); }

```

Comment:
possibly `return table.FieldNames;` ? although this would expose the field-names to abuse... slightly probematic

Reply:
edit: yeah, this (my code, not yours) also wouldn't take dead values into account; probably best to leave it alone

Reply:
Not a bad idea. I pretty much copied this from the existing IDictionary implementation, however so it really might be best to leave alone, for the sake of consistency.
(The only difference in the implementation is me skipping the .ToArray() at the end because for this interface it is enough to return an IEnumerable)